File: norintegrate-mcp/src/main/java/com/norintegrate/mcp/config/McpServerConfig.java
package com.norintegrate.mcp.config;

import com.norintegrate.mcp.tool.IntegrationGuideTool;
import com.norintegrate.mcp.tool.MunicipalitySearchTool;
import com.norintegrate.mcp.tool.ProcedureDetailTool;
import org.springframework.ai.tool.ToolCallbackProvider;
import org.springframework.ai.tool.method.MethodToolCallbackProvider;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class McpServerConfig {

  @Bean
  public ToolCallbackProvider toolCallbackProvider(
      IntegrationGuideTool integrationGuideTool,
      ProcedureDetailTool procedureDetailTool,
      MunicipalitySearchTool municipalitySearchTool) {
    return MethodToolCallbackProvider.builder()
        .toolObjects(integrationGuideTool, procedureDetailTool, municipalitySearchTool)
        .build();
  }
}

File: norintegrate-mcp/src/main/kotlin/com/norintegrate/mcp/config/McpServerConfig.kt
package com.norintegrate.mcp.config

import com.norintegrate.mcp.tool.IntegrationGuideTool
import com.norintegrate.mcp.tool.MunicipalitySearchTool
import com.norintegrate.mcp.tool.ProcedureDetailTool
import org.springframework.ai.tool.ToolCallbackProvider
import org.springframework.ai.tool.method.MethodToolCallbackProvider
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class McpServerConfig {
    @Bean
    fun toolCallbackProvider(
        integrationGuideTool: IntegrationGuideTool,
        procedureDetailTool: ProcedureDetailTool,
        municipalitySearchTool: MunicipalitySearchTool,
    ): ToolCallbackProvider =
        MethodToolCallbackProvider
            .builder()
            .toolObjects(integrationGuideTool, procedureDetailTool, municipalitySearchTool)
            .build()
}
